### ISSUE TITLE
Request: Git reset from the CLI should not remove files from the "Go to File..." dialog's "recently opened" section

### ISSUE THREAD

**Issue by DanKaplanSES** [microsoft/vscode#234006]
I consider this to be a minor UX improvement. 

Steps to Reproduce:

1. Create a new VSCode project in a git repo
1. Create a bunch of files with similar names. Open them all so they show up in the "Go to File..." dialog's "recently opened" section
1. Commit
1. Create a new directory in VSCode
1. Drag a file from step 2 to the new directory
1. Open a terminal to your project root
1. Run `git add .; git reset --hard`
1. Focus a different VS Code tab 
1. Search for the original file with the the "Go to File..." dialog (ctrl+p)

### Expected:
The original file appears in the "recently opened" section

### Actual:
The file will not be in the "recently opened" section

### Workaround:
You can still find the file in VS Code explorer. If you open it from there, the file will appear in the "Go to File..." dialog's "recently opened" section

## Note: 
1. Restarting VS Code does not make a difference
5. If you change step 8 to click on the deleted tab, the deleted file (i.e., the file with in invalid path) will appear in the "Go to File..." dialog thereafter.

**Comment by lszomoru** [comment]
@bpasero, any idea why a file would be removed from the "recently opened" section? Thanks!

**Comment by bpasero** [comment]
It is likely that the recently opened update from file delete events, is it possible that the actions here cause files to delete on disk?

**Comment by DanKaplanSES** [comment]
> is it possible that the actions here cause files to delete on disk?

Yes, step 7 deleted files on disk. I understand why this would cause problems. I'm requesting for VS Code to handle this situation gracefully.

**Comment by lszomoru** [comment]
@DanKaplanSES, thanks for the clarifications. I will defer to @bpasero as this is an area that he owns.
